scipy/stats/kde.py

```diff
@@ -559,12 +559,19 @@ def _compute_covariance(self):
         """Computes the covariance matrix for each Gaussian kernel using
         covariance_factor().
         """
+
         self.factor = self.covariance_factor()
         # Cache covariance and inverse covariance of the data
         if not hasattr(self, '_data_inv_cov'):
             self._data_covariance = atleast_2d(cov(self.dataset, rowvar=1,
                                                bias=False,
                                                aweights=self.weights))
+            # Small multiple of identity added to covariance matrix for
+            # numerical stability. Fixes gh-10205. For justification, see
+            # https://juanitorduz.github.io/multivariate_normal/ and primary
+            # source http://www.gaussianprocess.org/gpml/chapters/RWA.pdf
+            eps_I = 1e-14*np.eye(self.d)
```

Comment:
Please suggest a better magnitude of epsilon. For example, it should probably depend on the magnitude of the data, e.g.
```suggestion
            eps_I = 1e-14*np.max(self._data_covariance)*np.eye(self.d)
```

Reply:
The function would probably have a combination of the determinant of the matrix, number of elements in the matrix, and a blend of your existing formula. The rest is rule of thumb. Hopefully, it is not computationally expensive

Reply:
Sorry determinant is likely to be unstable too. So ignore the determinant.

Reply:
I don't know where this is going to be used but if it is going to be used in precision computations get the svd and invert sigma instead of pushing the data matrix.

Reply:
Thanks @ilayn. I don't know much about how this function is used, so i think I'll go ahead and leave it to someone who does.